src/reader.rs

```diff
@@ -942,6 +942,28 @@ pub fn reader_execute_readline_cmd(parser: &Parser, ch: CharEvent) {
     }
 }
 
+pub fn reader_showing_suggestion(parser: &Parser) -> bool {
+    if !is_interactive_session() {
+        return false;
+    }
+    if let Some(data) = current_data() {
+        let reader = Reader { parser, data };
+        // Work around the fact that autosuggestions happen asynchronously and we don't share state
+        // with the actual Reader used by the shell, so while a suggestion might have already been
+        // populated at the tty, we still need time for our background thread to populate one here.
+        // (We are blocking in a call to `commandline --showing-suggestion` here. We could use a
+        // condition variable instead of this loop, but this is incredibly far from the hot path,
+        // never called in a stock fish installation, and expected to generally never end up
+        // actually calling yield_now() except under CI or on really slow machines.)
+        while reader.autosuggestion.is_empty() && !reader.in_flight_autosuggest_request.is_empty() {
+            std::thread::yield_now();
```

Comment:
This won't have the intended effect - autosuggestions are determined in the background and are applied via a callback (see `autosuggest_completed`). Simply yielding to the scheduler won't cause this to run.

I suggest simply removing this unless something breaks without it, in which case we'll have to understand why.

Reply:
CI is broken without it.

Reply:
Yes, the completion runs on the same thread - aka the "main" thread. 

The CI [passed for me without this loop](https://github.com/ridiculousfish/fish-shell/actions/runs/9831107931). Indeed if the loop were ever entered, it would be an infinite, hang which I was able to reproduce with some testing.

CI failures here are probably the normal runners-are-slow stuff which can be addressed by increasing the sleep in the tests.

Please remove this bit and up the sleep if necessary. Otherwise this looks good.

Reply:
Thanks for checking it again. I can confirm it's passing now.